File: app/src/main/java/mega/privacy/android/app/Product.java
package mega.privacy.android.app;

import android.os.Parcel;
import android.os.Parcelable;

public class Product implements Parcelable{

	private long handle;
	private int level;
	private int months;
	private int storage;
	private int transfer;
	private int amount;
	private String currency;
	private boolean isBusiness;

	public Product(long handle, int level, int months, int storage, int transfer, int amount, String currency, boolean isBusiness) {
		this.handle = handle;
		this.level = level;
		this.months = months;
		this.storage = storage;
		this.transfer = transfer;
		this.amount = amount;
		this.currency = currency;
		this.isBusiness = isBusiness;
	}

	public long getHandle() {
		return handle;
	}

	public void setHandle(long handle) {
		this.handle = handle;
	}

	public int getLevel() {
		return level;
	}

	public void setLevel(int level) {
		this.level = level;
	}

	public int getMonths() {
		return months;
	}

	public void setMonths(int months) {
		this.months = months;
	}

	public int getStorage() {
		return storage;
	}

	public void setStorage(int storage) {
		this.storage = storage;
	}

	public int getTransfer() {
		return transfer;
	}

	public void setTransfer(int transfer) {
		this.transfer = transfer;
	}

	public int getAmount() {
		return amount;
	}

	public void setAmount(int amount) {
		this.amount = amount;
	}

	public String getCurrency() {
		return currency;
	}

	public void setCurrency(String currency) {
		this.currency = currency;
	}

	public boolean isBusiness() {
		return isBusiness;
	}

	public void setBusiness(boolean isBusiness) {
		this.isBusiness = isBusiness;
	}

	@Override
	public int describeContents() {
		// TODO Auto-generated method stub
		return 0;
	}

	@Override
	public void writeToParcel(Parcel out, int flags) {
		// TODO Auto-generated method stub
		out.writeLong(handle);
		out.writeInt(level);
		out.writeInt(months);
		out.writeInt(storage);
		out.writeInt(amount);
		out.writeInt(transfer);
		out.writeInt(isBusiness ? 1 : 0);
	}
}

File: app/src/main/java/mega/privacy/android/app/Product.kt
package mega.privacy.android.app

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * Data class containing all the required to present a product (PRO plan subscription) available.
 *
 * @property handle     Product handle
 * @property level      Product level (PRO I = 1, PRO II = 2, PRO III = 3, PRO LITE = 4, etc.)
 * @property months     Number of subscription months of the (1 for monthly or 12 for yearly)
 * @property storage    Amount of storage of the product
 * @property transfer   Amount of transfer quota of the product
 * @property amount     Amount or price of the product
 * @property currency   Currency of the product
 * @property isBusiness Flag to indicate if the product is business or not
 */
@Parcelize
data class Product(
    var handle: Long,
    var level: Int,
    var months: Int,
    var storage: Int,
    var transfer: Int,
    var amount: Int,
    var currency: String?,
    var isBusiness: Boolean,
) : Parcelable